File: JSTerm/JSKern/Process/ErrorThrow.m
//
//  ErrorThrow.m
//  JSTerm
//
//  Created by fridakitten on 14.03.25.
//

#import "ErrorThrow.h"

/*
 @Brief rewritten version of jsDoThrowError in ObjC
 */
JSValue* jsDoThrowError(JSContext *ctx, NSString *msg) {
    JSValue *error = [JSValue valueWithNewErrorFromMessage:msg inContext:ctx];
    [ctx setException:error];
    return error;
}

File: JSTerm/JSKern/Process/ErrorThrow.swift
//
//  ErrorThrow.swift
//  JSTerm
//
//  Created by fridakitten on 13.03.25.
//

import JavaScriptCore

/*
 @Brief a helper function for more debugging of JSBinaries
 */
func jsDoThrowError(_ error: String) -> JSValue {
    let error = JSValue(newErrorFromMessage: error, in: JSContext.current())!
    JSContext.current()?.exception = error
    return JSValue(undefinedIn: JSContext.current())!
}
